### PR TITLE
Clean up CLIMA initialization

### PR DESCRIPTION
**Pull request by kpamnany**
Removes the need to call `CLIMA.array_type()` at global scope which was previously necessary due to the dynamic definition of `_init_array()` for `CuArrays` causing a world age problem.

Once this is merged, `CLIMA.init()` should be the first function called. This also enables a `--disable-gpu` command line argument besides the `CLIMA_GPU=false` environment variable as well as a `--mpi-knows-cuda` command line argument that will be used to control https://github.com/climate-machine/CLIMA/pull/439.

### CODE REVIEW

File: src/Driver/Driver.jl [hunk 1]
```diff
@@ -18,25 +18,39 @@ using ..MoistThermodynamics
 using ..MPIStateArrays
 using ..TicToc
 
-const cuarray_pkgid = Base.PkgId(Base.UUID("3a865a2d-5b23-5a0f-bc46-62713ec82fae"), "CuArrays")
+mutable struct CLIMA_Settings
```
Comment (simonbyrne) [inline src/Driver/Driver.jl]: Can use `Base.@kwdef` to provide a keyword constructor and default values.

**Comment by kpamnany** [comment]
Will fix #655 before `r+`ing.

**Pull request merged**